src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/StorePal.cs

```diff
@@ -131,6 +131,35 @@ public static IStorePal FromSystemStore(string storeName, StoreLocation storeLoc
             throw new CryptographicException(message, new PlatformNotSupportedException(message));
         }
 
+        private static IStorePal FromCustomKeychainStore(string storeName, OpenFlags openFlags)
+        {
+            string storePath;
+
+            if (!IsValidStoreName(storeName))
+                throw new CryptographicException(SR.Format(SR.Security_InvalidValue, nameof(storeName)));
+                        
+            storePath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "Library", "Keychains", storeName + ".keychain");
```

Comment:
Hm. A concern I have with this approach is that on Windows and Linux the store name is case insensitive.  And on macOS it will now become case sensitive.  Perhaps this should normalize the store name to ToLowerInvariant()?

Reply:
I can do that. However it won't be possible to access third-party stores with non-lower-case names. My particular use case doesn't require that and it will still be possible using native APIs, so I am fine with that... (will write a unit test for it)